### PR TITLE
NMS-12692: Hostname resolution for flow queries

### PR DESCRIPTION
**Pull request by fooker**
This implements hostname querying for aggregated flows processed by nephron
and fixes hostname querying for row flows.

This is accompanied by https://github.com/OpenNMS/nephron/pull/3.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12692

### CODE REVIEW

File: features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/agg/AggregatedFlowQueryService.java [hunk 1]
```diff
@@ -84,22 +89,30 @@ public AggregatedFlowQueryService(JestClient client, IndexSelector indexSelector
 
     @Override
     public CompletableFuture<List<TrafficSummary<Conversation>>> getTopNConversationSummaries(int N, boolean includeOther, List<Filter> filters) {
-        return getTopNSummary(N, includeOther, filters, GroupedBy.EXPORTER_INTERFACE_CONVERSATION, Types.CONVERSATION );
+        return getTopNSummary(N, includeOther, filters, GroupedBy.EXPORTER_INTERFACE_CONVERSATION, Types.CONVERSATION);
```
Comment (j-white) [inline features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/agg/AggregatedFlowQueryService.java:92]: Conversations need to be mapped as well.

**Review by j-white — COMMENTED** [review]
Looks like this covers all the required functionality now - thanks for putting this together.

A couple comments:
1. Can we bring the `Types` class back? This is much nicer than using string literals everywhere.
2. Is it feasible to make one query to ES to resolve multiple hostnames instead of one query per host? When building a table of Top-10 conversations for example, this now requires 20 separate queries to ES with most of the same filter criteria.
3. Have we been able to see this work end-to-end w/ Nephron?

**Review by christianpape — APPROVED** [review]
Awesome!

**Comment by fooker** [comment]
@j-white 
>     1. Can we bring the `Types` class back? This is much nicer than using string literals everywhere.
Will try to. But it will be less fancy. It needs to store `String` only because most of the processing happens on the string type, not on the resulting object type.

>     2. Is it feasible to make one query to ES to resolve multiple hostnames instead of one query per host? When building a table of Top-10 conversations for example, this now requires 20 separate queries to ES with most of the same filter criteria.
Don't think there is a way to do this. AFAIK there is no way to specify a query for `host=lowerIp OR host=upperIp` while limiting the result to one answer each.

>     3. Have we been able to see this work end-to-end w/ Nephron?
I'm working on it...

**Review by christianpape — APPROVED** [review]
Looks good.

**Pull request merged**